packages/graph-engine/tests/data/input.test.ts

```diff
@@ -0,0 +1,93 @@
+import { AnySchema } from '../../src/schemas/index.js';
+import { Graph } from '../../src/graph/graph.js';
+import { Input } from '../../src/programmatic/input.js';
+import { Node } from '../../src/programmatic/node.js';
+import { Output } from '../../src/programmatic/output.js';
+import { beforeEach, describe, expect, test } from 'vitest';
+
+describe('Input', () => {
+	let graph: Graph;
+	let sourceNode: Node;
+	let targetNode: Node;
+	let sourceOutput: Output;
+	let targetInput: Input;
+
+	beforeEach(() => {
+		graph = new Graph();
+
+		sourceNode = new Node({
+			id: 'source',
+			graph: graph,
+			annotations: {}
+		});
+		sourceNode.addOutput('out', {
+			type: AnySchema
+		});
+		sourceOutput = sourceNode.outputs.out;
+		sourceOutput.set('source value');
+		graph.addNode(sourceNode);
+
+		targetNode = new Node({
+			id: 'target',
+			graph: graph,
+			annotations: {}
+		});
+		targetNode.addInput('in', {
+			type: AnySchema
+		});
+		targetInput = targetNode.inputs.in;
+		graph.addNode(targetNode);
+	});
+
+	test('connected input should return source output value', () => {
```

Comment:
Also test with the connect syntax eg `input1.outputs.value.connect(sub.inputs.inputs);` -